auth/oidc.go

```diff
@@ -439,3 +514,182 @@ func SetURLQueryParam(strURL, name, value string) (string, error) {
 	uri.RawQuery = rawQuery.Encode()
 	return uri.String(), nil
 }
+
+// Run runs the provider metadata sync task in background.
+func (op *OIDCProvider) StartDiscoverySync() error {
+	op.stopDiscoverySyncOnce.Do(func() {
+		op.stopDiscoverySync = make(chan struct{})
+	})
+	duration, err := op.runDiscoverySync()
+	if err != nil {
+		return err
+	}
+	go func() {
+		for {
+			select {
+			case <-time.After(duration):
+				duration, err = op.runDiscoverySync()
+				if err != nil {
+					base.Errorf("openID Connect provider discovery sync ends up in error: %v", err)
```

Comment:
I don't think we actually want to terminate the discoverySync loop on every error.  Imagine a case where the HTTP GET returns an error due to a temporary network issue - we wouldn't want to give up on future synchronization based on that.  I also think that error logging is too severe for that case - it would merit a warning at most.